service/src/main/java/io/camunda/service/DocumentServices.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.service;
+
+import io.camunda.search.clients.CamundaSearchClient;
+import io.camunda.service.security.auth.Authentication;
+import io.camunda.service.transformers.ServiceTransformers;
+import io.camunda.zeebe.broker.client.api.BrokerClient;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.time.ZonedDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+public class DocumentServices extends ApiServices<DocumentServices> {
+
+  // TODO: This is an in-memory implementation, replace it with a real document store
+  private static final String STORE_ID = "default";
+  private final Map<UUID, byte[]> documents = new HashMap<>();
+
+  public DocumentServices(final BrokerClient brokerClient, final CamundaSearchClient searchClient) {
+    this(brokerClient, searchClient, null, null);
+  }
+
+  public DocumentServices(
+      final BrokerClient brokerClient,
+      final CamundaSearchClient searchClient,
+      final ServiceTransformers transformers,
+      final Authentication authentication) {
+    super(brokerClient, searchClient, transformers, authentication);
+  }
+
+  @Override
+  public DocumentServices withAuthentication(final Authentication authentication) {
+    return new DocumentServices(brokerClient, searchClient, transformers, authentication);
+  }
+
+
+  public CompletableFuture<DocumentReferenceResponse> createDocument(
+      DocumentCreateRequest request) {
+    validateStoreId(request.storeId);
+    final String id = request.documentId != null ? request.documentId : UUID.randomUUID().toString();
+    final var contentInputStream = request.contentInputStream;
+    final byte[] content;
+    try {
+      content = contentInputStream.readAllBytes();
+      contentInputStream.close();
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to read document content", e);
```

Comment:
🔧 We have a `CamundaServiceException` you can use, if you like. Not a requirement, it won't add a lot of benefit for now here, but we might extend this in the future regarding automated handling of this type of exception or subtypes.

Reply:
But it could also be an `IllegalArgumentException`, I guess, since the user input is apparently off.